runtime/runtime/src/actions.rs

```diff
@@ -445,7 +445,28 @@ pub(crate) fn action_delete_account(
     result: &mut ActionResult,
     account_id: &AccountId,
     delete_account: &DeleteAccountAction,
+    current_protocol_version: ProtocolVersion,
 ) -> Result<(), StorageError> {
+    if current_protocol_version
+        >= PROTOCOL_FEATURES_TO_VERSION_MAPPING[&ProtocolFeature::DeleteActionRestriction]
```

Comment:
unrelated to PR, but feels like we can have a nicer API: `current_protocol_version.has_feature(ProtocolFeature::DeleteActionRestriction)`

Reply:
That is a good idea!